src/main/java/org/minions/devfund/lourdes/battleship/Ship.java

```diff
@@ -0,0 +1,141 @@
+package org.minions.devfund.lourdes.battleship;
+
+public abstract class Ship {
+    private int bowRow;
+    private int bowColumn;
+    protected int lenght;
+    private boolean horizontal;
+    protected boolean[] hit;
+
+    abstract String getShipType();
+
+    public boolean okToPlaceShipAt(int row, int column, boolean horizontal, Ocean ocean) {
+        if (row < 0 || row > ocean.getShipArray().length || column < 0 || column > ocean.getShipArray().length) {
+            return false;
+        }
+        return (verifyShipSpaceFree(row, column, horizontal, ocean, lenght) && verifySheepHorizontallyFree(row, column, horizontal, ocean));
+    }
+
+    private boolean verifySheepHorizontallyFree(int row, int column, boolean horizontal, Ocean ocean) {
+        boolean topRow = true;
+        boolean bottomRow = true;
+        boolean rightRow = true;
+        boolean leftRow = true;
+        if (row > 0 && row < ocean.getShipArray().length) {
+            topRow = verifyShipSpaceFree(row - 1, column - 1, horizontal, ocean, lenght + 2);
+            bottomRow = verifyShipSpaceFree(row + 1, column - 1, horizontal, ocean, lenght + 2);
+        }
+        if (column > 0 && column < ocean.getShipArray().length) {
+            rightRow = !ocean.isOccupied(row, lenght + 2);
+            leftRow = !ocean.isOccupied(row, column - 1);
+        }
+
+        return topRow && bottomRow && rightRow && leftRow;
+
+
+    }
+
+    private boolean verifyShipSpaceFree(int row, int column, boolean horizontal, Ocean ocean, int lenght) {
+        try {
+            for (int i = 0; i < lenght; i++) {
+                if (horizontal && ocean.isOccupied(row, column + i)) {
+                    return false;
+
+                } else if ((ocean.isOccupied(row + i, column))) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This branch's code block is the same as the block for the branch on line 41. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1871)